tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java

```diff
@@ -53,20 +54,24 @@ public Object execute(TransactionalExecutor business) throws Throwable {
         }
         Propagation propagation = txInfo.getPropagation();
         String previousXid = null;
+        String previousBranchType = null;
         try {
             switch (propagation) {
                 case NOT_SUPPORTED:
                     previousXid = RootContext.unbind();
                     return business.execute();
                 case REQUIRES_NEW:
                     previousXid = RootContext.unbind();
+                    previousBranchType = RootContext.unbindBranchType();
                     break;
                 case SUPPORTS:
                     if (StringUtils.isEmpty(RootContext.getXID())) {
                         return business.execute();
                     }
                     break;
```

Comment:
if xid not empty, need to deal with previousBranchType

Reply:
I aggree with you and now my strategy is: If propagation is RequiresNew, then suspend the  previousXid and branchtype  (whether previousXid is null or not), and desides whether to recover them or not in the  finally{} call  by judging whether the previous value is empty.

Reply:
My mistake, I mean SUPPORTS.